.github/workflows/build_rocks.yaml

```diff
@@ -55,24 +63,44 @@ jobs:
         with:
           script: |
             const path = require('path')
-            const workingDir = core.getInput('working-directory')
+            const inputs = ${{ toJSON(inputs) }}
+            const workingDir = inputs['working-directory']
+            const multiarch = inputs['multiarch-awareness']
             const rockcraftGlobber = await glob.create(
               path.join(workingDir, '**/rockcraft.yaml')
             )
             const rockPaths = []
             const images = []
+            const rockMetas = []
+            const defaultArch = 'amd64'
+            core.info(`Multiarch Awareness is ${multiarch ? "on" : "off" }`)
             for (const rockcraftFile of await rockcraftGlobber.glob()) {
               const rockPath = path.relative('.', path.dirname(rockcraftFile))
-              rockPaths.push(rockPath)
               core.info(`found rockcraft.yaml in ${rockPath}`)
               const fileHash = await glob.hashFiles(path.join(rockPath, '**'))
-              const rockName = (
-                await exec.getExecOutput('yq', ['.name', rockcraftFile])
+              const [rockName, rockVersion] = (
+                await exec.getExecOutput('yq', ['.name,.version', rockcraftFile])
+              ).stdout.trim().split("\n")
+              const platforms = (
+                await exec.getExecOutput('yq', ['.platforms | keys', '-o=json', rockcraftFile])
               ).stdout.trim()
-              const image = `${{ inputs.registry }}/${{ inputs.owner }}/${rockName}:${fileHash}`
-              core.info(`generate image name: ${image}`)
-              images.push(image)
+              if (multiarch && platforms) {
+                const arches = JSON.parse(platforms)
+                for (arch of arches) {
+                  const image = `${{ inputs.registry }}/${{ inputs.owner }}/${rockName}:${fileHash}-${arch}`
+                  core.info(`generate multi-arch image name: ${image}`)
+                  images.push(image)
+                  rockMetas.push({name: rockName, version: rockVersion, path: rockPath, arch: arch, image: image})
+                }
```

Comment:
when multiarch is engaged and `platforms` are described by `rockcraft.yaml`, push a separate image for each platform with the `arch` appended at the end of the tag